src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -601,5 +601,44 @@ public bool IsGenericDefinition
                 return HasInstantiation && IsTypeDefinition;
             }
         }
+
+        public uint GetSDIIndex()
+        {
+            switch (Category)
+            {
+                case TypeFlags.Void:
+                    return 0x0003;
+                case TypeFlags.Boolean:
+                    return 0x0030;
+                case TypeFlags.Char:
+                    return 0x0010;
```

Comment:
It might be nice to steal the enum from [here](https://github.com/Microsoft/cci/blob/126992bbf155dc8c6ba30cb0e9a12acc507c2579/PDBReaderAndWriter/PdbReader/CvInfo.cs#L213) so that the mapping is more reviewable.